### DISCUSSION TITLE
Where is `matx::non_owning` now?

### DISCUSSION THREAD

**Discussion by ZJUGuoShuai** [NVIDIA/MatX#384] [Q&A] [answered]
I used to create a `matx::tensor_t` from a pointer using `matx::non_owning` like [the document said](https://nvidia.github.io/MatX/creation.html?highlight=non_owning#matx-storage) and this #175 . But now it seems like `matx::non_owning` is removed? How can I create a `matx::tensor_t` from my pointer now?

**Comment by cliffburdick** [comment, marked as answer]
Hi @ZJUGuoShuai, non_owning is still available and is the default when you pass in your own pointer, whereas if you don't pass in your own pointer, the type defaults to owning. We did make a change several months ago to make this a runtime parameter rather than a template parameter. For example: 

```
auto make_tensor(T *data, const index_t (&shape)[RANK], bool owning = false) {
```
You would now do something like:
```
auto t = make_tensor(ptr, {1}, false);
auto t = make_tensor(ptr, {1}, true);
```
